### PR TITLE
bqam gcc arm

### PR DESCRIPTION
**Pull request by bqam-ublox**
Notes:
- Pull requests will not be accepted until the submitter has agreed to the [contributer agreement](https://github.com/ARMmbed/mbed-os/blob/master/CONTRIBUTING.md).
- This is just a template, so feel free to use/remove the unnecessary things

## Description

A few sentences describing the overall goals of the pull request's commits.

## Status

**READY/IN DEVELOPMENT/HOLD**

## Migrations

If this PR changes any APIs or behaviors, give a short description of what *API users* should do when this PR is merged.

YES | NO

## Related PRs

List related PRs against other branches:

branch | PR
------ | ------
other_pr_production | [link]()
other_pr_master | [link]()

## Todos

- [ ] Tests
- [ ] Documentation

## Deploy notes

Notes regarding the deployment of this PR. These should note any required changes in the build environment, tools, compilers and so on.

## Steps to test or reproduce

Outline the steps to test or reproduce the PR here.

### CODE REVIEW

**Comment by bqam-ublox** [comment]
@fahim-ublox kindly merge this in master

File: targets/TARGET_ublox/TARGET_R5030/device/TOOLCHAIN_GCC_ARM/startup_KM_app.S [hunk 1]
```diff
@@ -37,7 +37,7 @@
```
Comment (fahimalavi) [inline targets/TARGET_ublox/TARGET_R5030/device/TOOLCHAIN_GCC_ARM/startup_KM_app.S]: I have noticed that __HeapBase is utilized in this file but not defined in linker script, can you add it ?

File: targets/TARGET_ublox/TARGET_R5030/device/TOOLCHAIN_GCC_ARM/startup_KM_app.S [hunk 2]
```diff
@@ -37,7 +37,7 @@
 
 
     .syntax    unified
-    .arch    armv6-m
+    .arch    armv7-m
```
Comment (fahimalavi) [inline targets/TARGET_ublox/TARGET_R5030/device/TOOLCHAIN_GCC_ARM/startup_KM_app.S]: Cortex-M4 / M7 implements the Armv7E-M

File: targets/TARGET_ublox/TARGET_R5030/device/TOOLCHAIN_GCC_ARM/startup_KM_app.S (hunk 1 above)
Comment (bqam-ublox) [reply targets/TARGET_ublox/TARGET_R5030/device/TOOLCHAIN_GCC_ARM/startup_KM_app.S]: Added

**Review by fahimalavi — APPROVED** [review]

File: targets/TARGET_ublox/TARGET_R5030/device/TOOLCHAIN_GCC_ARM/startup_KM_app.S (hunk 2 above)
Comment (bqam-ublox) [reply targets/TARGET_ublox/TARGET_R5030/device/TOOLCHAIN_GCC_ARM/startup_KM_app.S]: Modified

**Review by fahimalavi — APPROVED** [review]

**Pull request merged**